src/pages/_app.tsx

```diff
@@ -35,6 +36,29 @@ export default function App({ Component, pageProps }: AppProps) {
   const router = useRouter();
   const previousRoute = usePreviousRoute();
 
+  // show success toast on sign in.
+  useEffect(() => {
+    if (router.query.s && router.query.p) {
+      toast({
+        title: 'Success',
+        description: `You've signed in with ${router.query.p}`,
+      });
+      const pathQuery = Object.entries(router.query).reduce(
+        (acc, [key, value]) => {
+          if (key !== 's' && key !== 'p') {
+            acc += `${key}=${value}&`;
+          }
+          return acc;
+        },
+        '',
+      );
+      router.replace({
+        query: pathQuery,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- only run on mount
```

Comment:
it runs when router is ready which is only once on mount...i am bad at documenting